osquery/utils/conversions/darwin/iokit.cpp

```diff
@@ -45,8 +45,11 @@ IOKitPCIProperties::IOKitPCIProperties(const std::string& compatible) {
 
   std::vector<std::string> vendor;
   boost::split(vendor, properties[prop_index++], boost::is_any_of(","));
+
   vendor_id = vendor[0].substr(3);
```

Comment:
Is this something that we need to check, or is this guaranteed by something earlier?

Reply:
@sharvilshah Thanks for catching this!

I wouldn't depend on the property always containing at least a comma; better be safe than sorry and check that the `vendor` is not empty first.

Reply:
@Smjert Thanks! Agree, let's do an extra check, updated!